### PR TITLE
redshift-wlroots: init at 1.12

### PR DESCRIPTION
**Pull request by matthias-t**
###### Motivation for this change

Make redshift available on sway and other wlroots based wayland compositors. Can be used in `services.redshift.package`. See also #57602. Please review thoroughly as this is my first contribution and I have no idea what I'm doing.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/redshift/default.nix
Comment (JohnAZoidberg) [inline pkgs/applications/misc/redshift/default.nix]: It's just a single patch. If it applies to the lastest release of the proper redshift, I'd rather do that with `patches = [ fetchpatch { url = "https://github.com/minus7/redshift/commit/eecbfedac48f827e96ad5e151de8f41f6cd3af66.diff"; sha256 = ""; } ];` rather than substituting the entire soruce tree.

**Review by JohnAZoidberg — CHANGES_REQUESTED** [review]
It doesn't seem like it builds the wayland parts, as the dependencies are missing (https://github.com/minus7/redshift/commit/eecbfedac48f827e96ad5e151de8f41f6cd3af66#diff-67e997bcfdac55191033d57a16d1408aR70) as well as the `--enable-wayland` configure flag.

You haven't tried to run it yet, right?

**Comment by matthias-t** [comment]
it seems to work on my machine. which dependencies am i missing?

File: pkgs/applications/misc/redshift/default.nix
Comment (matthias-t) [reply pkgs/applications/misc/redshift/default.nix]: it applies to the latest commit on master, which is 24 commits ahead of redshift v1.2

Comment (matthias-t) [reply pkgs/applications/misc/redshift/default.nix]: i agree that it's more elegant, but won't this be hard to maintain anyway in case redshift-wlroots falls behind and conflicts develop?

Comment (matthias-t) [reply pkgs/applications/misc/redshift/default.nix]: @JohnAZoidberg here you go, anyway

**Review by megheaiulian — APPROVED** [review]
I've tested this PR with `sway` by setting `services.redshift.package` and it works correctly.

**Comment by worldofpeace** [comment]
Is wayland support even added upstream?

If not, I don't think we should create a new package just for this feature.
It would be better to wait for the upstream maintainer to merge the change and
make a release.

**Comment by matthias-t** [comment]
> Is wayland support even added upstream?

no, see https://github.com/jonls/redshift/issues/55

> If not, I don't think we should create a new package just for this feature.
It would be better to wait for the upstream maintainer to merge the change and
make a release.

It's not exactly a feature, but a completely different backend, https://github.com/jonls/redshift/pull/663 is open since september 2018, there seems to be no movement upstream and it's the only way to keep my eyes sane on wayland in the meantime. Although if it isn't worth a new package that's fine since I'm using my fork anyway.

**Comment by worldofpeace** [comment]
Yeah, something like this would be a better fit for [NUR](https://github.com/nix-community/nur).
IIRC there's a lot of wayland users in nixos.

Closing as we've converged on this.

**Pull request closed without merging**
